### PR TITLE
Echo server and client

### PR DESCRIPTION
**Pull request by quinqu**
I was initially also using go's net/rpc package, but this version uses just the net/http package

*created this branch to get feedback on what I had already changed (and what is already on master), this version is just different from the first few commits.

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -13,16 +13,17 @@ import (
...
 )
 
 func main() {
 	kingpin.MustParse(app.Parse(os.Args[1:]))
 
 	var buf bytes.Buffer
+
 	message := strings.Join(*send, " ")
 	buf.WriteString(message)
 	host := "http://localhost:" + *port + "/echo"
```
Comment (wadells) [inline client/client.go:29]: Its weird to have a configurable port but a hardcoded host.  I'd recommend having both the host & port configurable (maybe in a single field).

File: server/echoserver.go [hunk 2]
```diff
@@ -16,21 +16,14 @@ func echo(w http.ResponseWriter, r *http.Request) {
 	fmt.Fprintf(w, newStr)
 }
 
-type Listener int
-
 func main() {
 
-	listener, err := net.Listen("tcp", ":8000") // open the connection
+	listener, err := net.Listen("tcp", ":8000")
```
Comment (wadells) [inline server/echoserver.go:21]: Should this also be configurable, since the client can choose a port?

File: server/echoserver.go [hunk 3]
```diff
@@ -16,21 +16,14 @@ func echo(w http.ResponseWriter, r *http.Request) {
...
 		log.Fatal(err)
 	}
 
-	// Serve accepts incoming HTTP connections on the listener,
-	// creating a new service goroutine for each. The service goroutines
-	// read requests and then call handler to reply to them.
-
 	http.HandleFunc("/echo", echo)
-
 	log.Println("Listening on localhost:8080")
```
Comment (wadells) [inline server/echoserver.go:27]: This log statement lies about the port number.

**Review by wadells — COMMENTED** [review]
Its difficult to review this, since it is off of unreviewed content already in master.  I'd recommend saving your current work in a branch, and then resetting master to 249c68f0b27c2b03965583a20949394fae1495a0.  Force push this, and you can PR all the changes.

File: server/echoserver.go (hunk 3 above)
Comment (quinqu) [reply server/echoserver.go:27]: Oh silly typo

File: client/client.go (hunk 1 above)
Comment (quinqu) [reply client/client.go:29]: I did not know that was weird, but good to know. I'll add that in

**Pull request closed without merging**